packages/adblocker/src/filters/network.ts

```diff
@@ -1535,11 +1524,64 @@ function setNetworkMask(mask: number, m: number, value: boolean): number {
   return clearBit(mask, m);
 }
 
+function checkIsAlphabet(charCode: number) {
+  return charCode >= 97 && charCode <= 122;
+}
+
+function checkIsHostnameToken(charCode: number) {
+  return (
+    checkIsAlphabet(charCode) ||
+    (charCode >= 48 && charCode <= 57) || // Numerics
+    charCode === 45 // Hyphen
+  );
+}
+
+function checkIsHostname(filter: string, start: number, end: number): boolean {
```

Comment:
I think it'd be worth reusing the function from `tldts` for hostname validation.